plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/DebugNativeAppEngineStandardProjectTest.java

```diff
@@ -89,11 +91,25 @@ public void testDebugLaunch() throws Exception {
 
     SWTBotTreeItem testProject = SwtBotProjectActions.selectProject(bot, "testapp_java8");
     assertNotNull(testProject);
-    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, () -> {
-      System.out.println("==== menu items of Run > Debug As ====");
-      bot.menu("Run").menu("Debug As").menuItems().forEach(System.out::println);
-      bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
-    });
+    SwtBotTestingUtilities.performAndWaitForWindowChange(
+        bot,
+        () -> {
+          bot.waitUntil(
```

Comment:
I think it's more readable to call this outside the runnable, before `performAndWaitForWindowChange()`. Also conceptually, the wait is an extra step that is independent of taking the click action.

Reply:
The key here is that we perform some action (Run As > Run On Server) and then wait for the window change (the dialog to appear).  If we perform the action outside then the dialog may appear before we wait.

Reply:
Sorry, what I meant is

```java
    bot.waitUntil(new DefaultCondition() {
      @Override
      public boolean test() throws Exception {
        return Iterables.any(..., text -> "1 Debug on Server".equals(text));
      }
     ...
    });
    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, () -> {
      bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
    });
```

Looking into the code of `performAndWaitForWindowChange()`, there seems no real difference between this and the code in he PR.

Reply:
That should work.